packages/gasket-request/README.md

```diff
@@ -0,0 +1,130 @@
+# @gasket/request
+
+The purpose of this package is to provide a consistent request object for Gasket
+plugins and apps to use, regardless of the request handling framework.
+
+## Installation
+
+#### Existing apps
+
+```shell
+npm install @gasket/request
+```
+
+## Functions
+
+### GasketRequest
+
+The `GasketRequest` class is a representation of a request object that can be
+used by Gasket plugins and apps. It is a consistent shape that can be used
+across different request handling frameworks.
+
+A `GasketRequest` object has the following properties:
+- `headers` - (object) Request headers
+- [`cookies`] - (object) Request cookies
+- [`query`] - (object) Query parameters
+- [`path`] - (string) Request path
+
+A `GasketRequest` can be created from a Node `IncomingMessage` object, an Express
+`Request` object, or a Next.js `NextRequest` object, amongst others.
+
+```js
+import { GasketRequest } from '@gasket/request';
+
+export default async function handler(req, res) {
+  const gasketRequest = new GasketRequest.make(req);
+  // use gasketRequest
+}
+```
+
+You can also assemble a `GasketRequest` object from parts of a request object.
+The only required property is the `headers`, which is used as a caching key to
+ensure the same `GasketRequest` instance is used throughout the request lifetime.
+
+```js
+import { GasketRequest } from '@gasket/request';
+
+const headers = {
+  'x-example': 'example'
+};
+
+const staticGasketRequest = new GasketRequest.make({ headers });
+````
+
+### makeGasketRequest
+
+Another way to create a `GasketRequest` object is to use the `makeGasketRequest`.
```

Comment:
Do we need two ways to create these? Maybe we should just have one blessed gasket creation function in the module interface.

Reply:
Yea, we don't need 2 ways. Probably wisest to only have one for discussion and debugging.